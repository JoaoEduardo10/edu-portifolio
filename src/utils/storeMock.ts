import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

const store = mockStore({
  menu: {
    menuOn: false,
    colorOn: true,
  },
  completeProject: {
    id: "1234",
    project: false,
  },
});

export { store };
