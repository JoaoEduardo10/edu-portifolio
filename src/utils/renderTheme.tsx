/* eslint-disable @typescript-eslint/no-explicit-any */
import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../style/themes";
import { Provider } from "react-redux";
import { GlobalStyles } from "../style/globalStyles";
import { ToolkitStore } from "@reduxjs/toolkit/dist/configureStore";
import { store } from "./storeMock";

const renderTheme = (children: React.ReactNode, storeParams?: ToolkitStore) => {
  const isStore = storeParams ?? store;
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={isStore}>
        {children}
        <GlobalStyles />
      </Provider>
    </ThemeProvider>
  );
};

export { renderTheme };
