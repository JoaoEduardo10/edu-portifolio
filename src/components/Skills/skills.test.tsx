import React from "react";
import { screen } from "@testing-library/react";
import { Skills } from ".";
import { renderTheme } from "../../utils/renderTheme";
import { describe, it } from "vitest";

describe("<Skills />", () => {
  it("should a render skills component", () => {
    renderTheme(
      <Skills text="test">
        <div aria-label="icon">test</div>
      </Skills>
    );

    const conteinerSkil = screen.getByLabelText("Conteiner Skill");
    const heading = screen.getByRole("heading", { name: "test" });
    const icon = screen.getByLabelText("icon");

    expect(conteinerSkil).toBeInTheDocument();
    expect(heading).toBeInTheDocument();
    expect(icon).toBeInTheDocument();

    expect(conteinerSkil).toMatchSnapshot();
  });
});
