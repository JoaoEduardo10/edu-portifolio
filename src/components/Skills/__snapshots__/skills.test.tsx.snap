// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Skills /> > should a render skills component 1`] = `
.c0 {
  position: relative;
  background-color: #212121;
  width: 12rem;
  height: 12rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0 3rem 0 0;
}

.c0::after {
  content: '';
  display: block;
  position: absolute;
  width: 100%;
  height: 0.3rem;
  background-image: linear-gradient(90deg,#8257E6 0%,#A40EE4 92.32%);
  bottom: -0.1rem;
  left: 0;
}

.c0 h3 {
  font-size: 1.6rem;
  margin-bottom: 1.6rem;
  color: #8257E6;
}

.c0 svg {
  font-size: 6rem;
  color: #8257E6;
}

.c0:hover {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

<div
  aria-label="Conteiner Skill"
  class="c0"
>
  <h3>
    test
  </h3>
  <div
    aria-label="icon"
  >
    test
  </div>
</div>
`;
